src/macros.rs

```diff
@@ -30,11 +30,13 @@
 #[macro_export]
 #[clippy::format_args]
 macro_rules! log {
-    // log!(target: "my_target", Level::Info, key1:? = 42, key2 = true; "a {} event", "log");
-    (target: $target:expr, $lvl:expr, $($key:tt $(:$capture:tt)? $(= $value:expr)?),+; $($arg:tt)+) => ({
+    // log!(logger: my_logger, target: "my_target", Level::Info, key1:? = 42, key2 = true; "a {} event", "log");
+    (logger: $logger:expr, target: $target:expr, $lvl:expr, $($key:tt $(:$capture:tt)? $(= $value:expr)?),+; $($arg:tt)+) => ({
+        let logger = $logger;
```

Comment:
This has the semantic of executing `$logger` unconditionally, do we want to move this inside the `if` statement below to make its execution conditionally like other arguments?